### PR TITLE
fix(BaseDropdown): border styles & dedupe

### PR DESCRIPTION
**Pull request by zettca**
Components using `HvBaseDropdown` are handling their own styles (border, background, icon alignment, etc.).
This is leading to duplicated code, and sometimes missing styles that could be part of `HvBaseDropdown`.

This PR dedupes the styles of components using `HvBaseDropdown`, moving the styles to it.
It's mostly a refactor, but it also affects (fixes) some components with missing styles and the border color to `secondary_80`:

Examples:
- Missing Pagination border when opened and wrong color when closed
image

### CODE REVIEW

File: packages/core/src/components/ColorPicker/ColorPicker.styles.tsx [hunk 1]
```diff
@@ -28,29 +29,15 @@ export const styles: Partial<
     display: "flex",
     justifyContent: "center",
     padding: theme.colorPicker.panelPadding,
-    border: theme.colorPicker.panelBorder,
-    borderRadius: theme.colorPicker.panelBorderRadius,
```
Comment (plagoa) [inline packages/core/src/components/ColorPicker/ColorPicker.styles.tsx:32]: I think these two props can now be remove from the theme as they're no longer used:
https://github.com/lumada-design/hv-uikit-react/blob/master/packages/styles/src/types.ts#L359
https://github.com/lumada-design/hv-uikit-react/blob/master/packages/styles/src/theme.ts#L352
https://github.com/lumada-design/hv-uikit-react/blob/master/packages/styles/src/themes/ds3.ts#L568
https://github.com/lumada-design/hv-uikit-react/blob/master/packages/styles/src/themes/ds5.ts#L511

Comment (zettca) [reply packages/core/src/components/ColorPicker/ColorPicker.styles.tsx:32]: that's right 👍

File: packages/core/src/components/DatePicker/DatePicker.styles.tsx [hunk 2]
```diff
@@ -28,28 +28,16 @@ export const styles: Partial<
   dropdown: {
     display: "block",
   },
-  panel: {
-    border: theme.datePicker.panelBorder,
-    backgroundColor: theme.datePicker.panelBackgroundColor,
```
Comment (plagoa) [inline packages/core/src/components/DatePicker/DatePicker.styles.tsx:33]: These three props can be removed from the theme:
- panelBorder
- panelBackgroundColor
- dropdownHeaderOpenBorder

File: packages/core/src/components/Dropdown/Dropdown.styles.tsx [hunk 3]
```diff
@@ -62,13 +62,6 @@ export const StyledDropdown = styled(
         border: theme.dropdown.dropdownHeaderInvalidBorder,
       },
     },
-
-    [`& .${dropdownClasses.dropdownHeaderOpen}`]: {
-      border: theme.dropdown.dropdownHeaderOpenBorder,
```
Comment (plagoa) [inline packages/core/src/components/Dropdown/Dropdown.styles.tsx:67]: No longer needed theme props:
- dropdownHeaderOpenBorder
- headerBorderHover
- listBackgroundColor
- listBorderRadius

File: packages/core/src/components/FilterGroup/FilterContent/FilterContent.styles.tsx [hunk 4]
```diff
@@ -19,11 +19,8 @@ export const styles: Partial<
   },
   root: {
     width: 640,
-    backgroundColor: theme.colors.atmo1,
     maxHeight: "calc(500px - 75px)",
     minHeight: "calc(370px - 75px)",
-    border: theme.filterGroup.panelsBorder,
-    borderBottom: theme.filterGroup.panelsBorderBottom,
```
Comment (plagoa) [inline packages/core/src/components/FilterGroup/FilterContent/FilterContent.styles.tsx:26]: No longer needed theme props:
- panelsBorder
- panelsBorderBottom
- actionBarBorder
- actionBarBorderTop

File: packages/core/src/components/Pagination/Select.styles.tsx [hunk 5]
```diff
@@ -1,25 +1,19 @@
...
-      backgroundColor: "transparent",
-      borderColor: "transparent",
-    },
-    [`& .${baseDropdownClasses.headerOpen}`]: {
-      backgroundColor: theme.colors.atmo1,
-      borderColor: theme.colors.atmo1,
-    },
-    [`& .${baseDropdownClasses.selection}`]: {
-      borderRadius: theme.pagination.pageSizeBorderRadius,
-      border: `1px solid ${theme.pagination.pageSizeBorderColor}`,
```
Comment (plagoa) [inline packages/core/src/components/Pagination/Select.styles.tsx:22]: No longer needed theme props:
- pageJumpTextAlign

File: packages/core/src/components/Pagination/Select.tsx [hunk 6]
```diff
@@ -1,27 +1,21 @@
...
-} from "../index";
-import { StyledRoot } from "./Select.styles";
+import { HvBaseDropdown, HvSelectionList, HvListItem, HvPanel } from "..";
+import { useClasses } from "./Select.styles";
 
 export const Option = ({ children, ...others }) => (
   <HvListItem {...others}>{children}</HvListItem>
 );
 
 const HvSelect = ({
```
Comment (MEsteves22) [inline packages/core/src/components/Pagination/Select.tsx:9]: Noticed the following in the DS3, the extension seems to be bigger:

Screenshot 2023-06-16 at 12 05 04

File: packages/core/src/components/Dropdown/Dropdown.styles.tsx [hunk 7]
```diff
@@ -37,7 +37,7 @@ export const StyledDropdown = styled(
     $readOnly?: boolean;
   }) => ({
```
Comment (MEsteves22) [inline packages/core/src/components/Dropdown/Dropdown.styles.tsx:38]: The dropdown in DS3 doesn't have the right styling:

Screenshot 2023-06-16 at 12 13 18

Before:

Screenshot 2023-06-16 at 12 16 07

File: packages/core/src/components/ColorPicker/ColorPicker.tsx [hunk 8]
```diff
@@ -276,14 +273,9 @@ export const HvColorPicker = ({
                     className={cx(
                       colorPickerClasses.headerColorIcon,
                       css(styles.headerColorIcon),
-                      css({
-                        "& svg:last-child": {
-                          fill: color,
-                        },
-                      }),
```
[the remark this answers is not in the document]
Comment (zettca) [reply packages/core/src/components/ColorPicker/ColorPicker.tsx:283]: Can't recall why I touched this code, but I totally missed this 🤦
As the icon is `Checkbox` and the border is the second svg "element", I re-added leveraging the `color` prop with `[color, "transparent"]`

Let me know what you think!

File: packages/core/src/components/Dropdown/Dropdown.styles.tsx (hunk 7 above)
Comment (zettca) [reply packages/core/src/components/Dropdown/Dropdown.styles.tsx:38]: Fixed. `HvDrodown` still had some unnecessary overrides

File: packages/core/src/components/Pagination/Select.tsx (hunk 6 above)
Comment (zettca) [reply packages/core/src/components/Pagination/Select.tsx:9]: I can't reproduce this, in any browser 🤔

Comment (MEsteves22) [reply packages/core/src/components/Pagination/Select.tsx:9]: Sorry, it depends on the number selected and it was already happening in master but I didn't notice the first time around 😅 My bad

File: packages/core/src/components/ColorPicker/ColorPicker.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (MEsteves22) [reply packages/core/src/components/ColorPicker/ColorPicker.tsx:283]: Looks good to me 🙏

**Review by MEsteves22 — APPROVED** [review]

**Pull request merged**